pkg/splunk/enterprise/indexercluster.go

```diff
@@ -1069,3 +1098,139 @@ func RetrieveCMSpec(ctx context.Context, client splcommon.ControllerClient, cr *
 
 	return "", nil
 }
+
+func getIndexerClusterSortedSiteList(ctx context.Context, c splcommon.ControllerClient, ref corev1.ObjectReference, indexerList enterpriseApi.IndexerClusterList) (enterpriseApi.IndexerClusterList, error) {
+
+	namespaceList := enterpriseApi.IndexerClusterList{}
+
+	for _, v := range indexerList.Items {
+		if v.Spec.ClusterManagerRef == ref {
+			namespaceList.Items = append(namespaceList.Items, v)
+		}
+	}
+
+	sort.SliceStable(namespaceList.Items, func(i, j int) bool {
+		return getSiteName(ctx, c, &namespaceList.Items[i]) < getSiteName(ctx, c, &namespaceList.Items[j])
+	})
+
+	return namespaceList, nil
+}
+
+func getSiteName(ctx context.Context, c splcommon.ControllerClient, cr *enterpriseApi.IndexerCluster) string {
+	defaults := cr.Spec.Defaults
+	pattern := `site:\s+(\w+)`
```

Comment:
Nit: may be a constant with a comment for a example?

Reply:
we will fix this in next cycle.

Reply:
Just a thought , i think its not a good idea to keep this as constant, as this is only used in this function to get site name. i have added comment.